components/lcd_touch/esp_lcd_touch_gt911/esp_lcd_touch_gt911.c

```diff
@@ -181,15 +185,32 @@ static esp_err_t esp_lcd_touch_gt911_read_data(esp_lcd_touch_handle_t tp)
     /* Any touch data? */
     if ((buf[0] & 0x80) == 0x00) {
 #if (CONFIG_ESP_LCD_TOUCH_MAX_BUTTONS > 0)
-        tp->data.button[0].status = 0x00;
+        for (i = 0; i < CONFIG_ESP_LCD_TOUCH_MAX_BUTTONS; i++) {
```

Comment:
when all the touch button liftoff, we'll not enter the case `else if ((buf[0] & 0x10) == 0x10) {` 
So I think we need this to clean the button event.

Reply:
Are you sure? I think, that this bit is set, when some button state changed. Or is it only set, when some button is true and thus bit is not set, when button state is false?

Reply:
> Are you sure? I think, that this bit is set, when some button state changed. Or is it only set, when some button is true and thus bit is not set, when button state is false?

I've double check it.
when btn pressed, it'll report event 0x90`(once)` 0x00 0x00 0x00`(Infinite loop)`
when btn release, it'll  report event 0x80`(once)` 0x00 0x00 0x00`(Infinite loop)`

Maybe we shoule clean the event in the 0x80 event ?

Reply:
The event 0x80 is same for coordinates, then it should be instead of `else` ==> `else if((buf[0] & 0x80) == 0x80)` and there should be clear of buttons, what do you think?

Reply:
> The event 0x80 is same for coordinates, then it should be instead of `else` ==> `else if((buf[0] & 0x80) == 0x80)` and there should be clear of buttons, what do you think?

Yeah, it's better, I've updated it.